pkgs/development/mobile/androidenv/repo.json

```diff
@@ -15331,15 +18252,91 @@
           }
         ],
         "displayName": "Android Emulator",
-        "last-available-day": 19954,
+        "last-available-day": 19954,
+        "license": "android-sdk-preview-license",
+        "name": "emulator",
+        "path": "emulator",
+        "revision": "35.2.5",
+        "revision-details": {
+          "major:0": "35",
+          "micro:2": "5",
+          "minor:1": "2"
+        },
+        "type-details": {
+          "element-attributes": {
+            "xsi:type": "ns5:genericDetailsType"
+          }
+        }
+      },
+      "35.3.9": {
+        "archives": [
+          {
+            "os": "linux",
+            "sha1": "b7c8b4ecf0b0ccead3d97aeeb954cafcbdd51dc5",
+            "size": 297018503,
+            "url": "https://dl.google.com/android/repository/emulator-linux_x64-12594403.zip"
+          },
+          {
+            "os": "macosx",
+            "sha1": "e28710c310e310965bd81ef132876c33854b13bb",
+            "size": 311185949,
+            "url": "https://dl.google.com/android/repository/emulator-darwin_aarch64-12594403.zip"
+          },
+          {
+            "os": "windows",
+            "sha1": "21c1a3e980d7df5f01dcdb2b0bdbaa721dceb41f",
+            "size": 419749876,
+            "url": "https://dl.google.com/android/repository/emulator-windows_x64-12594403.zip"
+          }
+        ],
+        "displayName": "Android Emulator",
+        "last-available-day": 20056,
+        "license": "android-sdk-license",
+        "name": "emulator",
+        "path": "emulator",
+        "revision": "35.3.9",
+        "revision-details": {
+          "major:0": "35",
+          "micro:2": "9",
+          "minor:1": "3"
+        },
+        "type-details": {
+          "element-attributes": {
+            "xsi:type": "ns5:genericDetailsType"
+          }
+        }
+      },
+      "35.4.2": {
+        "archives": [
+          {
+            "os": "linux",
+            "sha1": "c09ec79c1a8a79ab2ce51ba2c35e91de65702efe",
+            "size": 304191081,
+            "url": "https://dl.google.com/android/repository/emulator-linux_x64-12658423.zip"
+          },
+          {
+            "os": "macosx",
+            "sha1": "88d045da40b76e9ff0893c4b477866bf0119ecd1",
+            "size": 320263693,
+            "url": "https://dl.google.com/android/repository/emulator-darwin_aarch64-12658423.zip"
+          },
```

Comment:
I believe this will break things for people on `x86_64-darwin` 

in addition to updating the repo urls, we also need to adapt the script which generates `repo.json` to make a distinction between `x86_64-darwin` and `darwin_aarch64` and finally adapt the nix code [here](https://github.com/NixOS/nixpkgs/blob/74831549826cb12b58952e92b6827e2e2110b25f/pkgs/development/mobile/androidenv/compose-android-packages.nix#L30-L31) (and everywhere where the `os` variable is used), to choose the correct architecture for a package.

Reply:
Interesting! You are right! Let me check! We may need to iterate like this multiple times, since I don't have access to an aarch64 CPU, right now!

Reply:
Sure. I’m a Linux user myself but I can test this out on macOS, both aarch64 and x86, on github actions 

thanks!

Reply:
@adrian-gierakowski please review and test! Thanks!

Reply:
we're getting problem, but I think we still have some issues:
1. we are missing x86_64-darwin versions of the emulator in [repo.json.](https://github.com/NixOS/nixpkgs/blob/a7927a90b7c16703a01189693639366ec4f31132/pkgs/development/mobile/androidenv/repo.json#L18857-L18879) (compare that to [android-nixpkgs](https://github.com/tadfisher/android-nixpkgs/blob/6fc652b651d3cd2c7d40f1f6a40630041cc0687a/channels/beta/default.nix#L3005-L3021) which has 4 entries).
2. seem like "arch" is missing for some versions of emulator (and maybe other packades) in `repo.json`, for example it's present [here](https://github.com/NixOS/nixpkgs/blob/a7927a90b7c16703a01189693639366ec4f31132/pkgs/development/mobile/androidenv/repo.json#L18737-L18760) but missing [here](https://github.com/NixOS/nixpkgs/blob/a7927a90b7c16703a01189693639366ec4f31132/pkgs/development/mobile/androidenv/repo.json#L18700-L18719)

consequence of point 1 is that we get correct binary on aarch64-darwin as can be seen in [this test run](https://github.com/rhinofi/android-nixpkgs-test/actions/runs/12096436144/job/33730732869)) 

![image](https://github.com/user-attachments/assets/86cf88f5-ba71-423b-aa6a-ab2d218deea9)


but get a build (or rather evaluation) failure x86_64-dawrwin (see [here](https://github.com/rhinofi/android-nixpkgs-test/actions/runs/12096436145/job/33730732850))
 
consequence of point 2 is that for some versions of emulator (35.1.4) we're get the old behaviour

Reply:
@adrian-gierakowski Sorry for late reply! I pushed a commit to address these issues. Let me know if this one works :D